vcd/resource_vcd_catalog_media.go

```diff
@@ -120,13 +132,23 @@ func resourceVcdMediaCreate(ctx context.Context, d *schema.ResourceData, meta in
 
 	vcdClient := meta.(*VCDClient)
 
-	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
-	if err != nil {
-		return diag.Errorf(errorRetrievingOrg, err)
-	}
-
+	var catalog *govcd.Catalog
+	var err error
 	catalogName := d.Get("catalog").(string)
-	catalog, err := adminOrg.GetCatalogByName(catalogName, false)
+	catalogId := d.Get("catalog_id").(string)
+	if catalogId == "" {
+		var adminOrg *govcd.AdminOrg
+		if catalogName == "" {
```

Comment:
I think this one can't happen anymore thanks to `ExactlyOneOf` if I'm not wrong

Reply:
Yes. Removed